cmd/replay/main.go

```diff
@@ -286,8 +286,8 @@ func main() {
 			return wavmio.ReadInboxMessage(batchNum), nil
 		}
 
-		// Handle the various pre-conditions and panics that can happen before we should enter the espresso logic in the validators STF
-		validatingAgainstEspresso, panicHandler := handleEspressoPreConditions(message, chainConfig.ArbitrumChainParams.EnableEspresso)
+		// Handle the various pre-conditions if the message is an Espresso message
+		validatingAgainstEspresso := arbos.IsEspressoMsg(message.Message)
```

Comment:
I'm not convince it's not correct to decide solely on the content of the message. We do want to enable the Espresso integration when the ArbOS upgrade is enabled. In #215 this is done via `chainConfig.ArbitrumChainParams.EnableEspresso`.

Reply:
makes sense, I guess then we will have to keep our fork. I will add the `chainConfig.ArbitrumChainParams.EnableEspresso` condition here as well